pkgs/applications/misc/gcstar/default.nix

```diff
@@ -0,0 +1,79 @@
+{stdenv, fetchFromGitLab, bash, perl, gtk2, perlPackages, wrapGAppsHook}:
+
+stdenv.mkDerivation rec {
+  pname = "gcstar";
+  version = "1.7.2";
+
+  src = fetchFromGitLab {
+      owner = "Kerenoc";
+      repo = "GCstar";
+      rev = "v${version}";
+      sha256 = "1vqfff33sssvlvsva1dflggmwl00j5p64sn1669f9wrbvjkxgpv4";
+    };
+
+  enableParallelBuilding = false;
+
+  perlDeps = [
+    perl
+    perlPackages.ArchiveZip
+    perlPackages.DateCalc
+    perlPackages.DateTimeFormatStrptime
+    perlPackages.Glib
+    perlPackages.Gtk2
+    perlPackages.GD
+    perlPackages.GDGraph
+    perlPackages.GDText
+    perlPackages.HTMLParser
+    perlPackages.JSON
+    perlPackages.ImageExifTool
+    perlPackages.librelative
+    perlPackages.LWPUserAgent
+    perlPackages.LWPProtocolHttps
+    perlPackages.MP3Info
+    perlPackages.MP3Tag
+    perlPackages.NetFreeDB
+    perlPackages.OggVorbisHeaderPurePerl
+    perlPackages.Pango
+    perlPackages.XMLSimple
+    perlPackages.XMLParser ];
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+  buildInputs = perlDeps;
+  propagatedBuildInputs = perlDeps;
+
+  installPhase = ''
+    cd gcstar
```

Comment:
When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`.